### PR TITLE
feat(generator): separate page for retry policy overrides

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #10157

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11950)

### CODE REVIEW

File: generator/internal/scaffold_generator.cc [hunk 1]
```diff
@@ -838,6 +834,97 @@ client library to change this default.
   printer.Print(variables, kText);
 }
 
+void GenerateOverrideRetryPoliciesPage(
+    std::ostream& os, std::map<std::string, std::string> const& variables) {
+  auto constexpr kText = R"""(/*!
+@page $library$-override-retry Override Retry, Backof, and Idempotency Policies
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:840]: s/Backof/Backoff/

File: generator/internal/scaffold_generator.cc [hunk 2]
```diff
@@ -838,6 +834,97 @@ client library to change this default.
...
+
+The `*RetryPolicyOption` controls:
+
+- Which errors are to be treated as transient errors.
+- How long will the library will keep retrying transient errors.
+
+You can provide your own class for this option. The library also provides two
+built-in policies:
+
+- `*LimitedErrorCountRetryPolicy`: stops retrying after an specified number
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:872]: s/an specified/a specified/

File: generator/internal/scaffold_generator.cc [hunk 3]
```diff
@@ -838,6 +834,97 @@ client library to change this default.
...
+  of transient errors.
+- `*LimitedTimeRetryPolicy`: stops retrying after an specified time.
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:874]: s/an specified/a specified/

File: generator/internal/scaffold_generator.cc [hunk 4]
```diff
@@ -838,6 +834,97 @@ client library to change this default.
...
+
+Note that a library may have more than one version of these classes. Their name
+matches the `*Client` and `*Connection` object they are intended to be used
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:877]: s/Their name matches/Their names match/

File: generator/internal/scaffold_generator.cc [hunk 5]
```diff
@@ -838,6 +834,97 @@ client library to change this default.
...
+
+The only built-in backoff policy is
+[`ExponentialBackoffPolicy`](@ref google::cloud::ExponentialBackoffPolicy).
+This class implements a truncated exponential backoff algorithm, with jitter.
+In summary, it doubles the current backoff time after each failure. The actual
+backoff time for an RPC is chosen at random, but never exceeds the current
+backoff. The current backoff is doubled after each failure, but never exceeds
+(or is "truncated") if it reaches a prescribed maximum.
+
+@section $library$-override-retry-idempotency-policy Controlling the which operations are retryable
```
Comment (dbolduc) [inline generator/internal/scaffold_generator.cc:896]: s/the which/which/

File: generator/internal/scaffold_generator_test.cc [hunk 6]
```diff
@@ -335,6 +337,16 @@ TEST_F(ScaffoldGenerator, OverrideEndpointPage) {
 )""")));
 }
 
+TEST_F(ScaffoldGenerator, OverrideRetryPoliciesPage) {
+  auto const vars = ScaffoldVars(path(), MockIndex(), service(), false);
+  std::ostringstream os;
+  GenerateOverrideRetryPoliciesPage(os, vars);
+  auto const actual = std::move(os).str();
+  EXPECT_THAT(actual, AllOf(HasSubstr(R"""(
+@page test-override-retry Override Retry, Backof, and Idempotency Policies
```
Comment (dbolduc) [inline generator/internal/scaffold_generator_test.cc:346]: s/Backof/Backoff/

File: generator/internal/scaffold_generator.cc (hunk 1 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: Done

File: generator/internal/scaffold_generator.cc (hunk 2 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: Done

File: generator/internal/scaffold_generator.cc (hunk 3 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: Done

File: generator/internal/scaffold_generator.cc (hunk 4 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: Done

File: generator/internal/scaffold_generator.cc (hunk 5 above)
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: Done

File: generator/internal/scaffold_generator_test.cc (hunk 6 above)
Comment (coryan) [reply generator/internal/scaffold_generator_test.cc]: Done

**Comment by coryan** [comment]
PTAL

**Review by dbolduc — APPROVED** [review]

**Pull request merged**